### PR TITLE
Add dbpath and env options

### PR DESCRIPTION
**Pull request by linxGnu**
This PR:
- Remove some duplicated tests to avoid errors while executing tests on local machine
- Moving all tests in `db.rs` to `tests/test_db.rs`, as well as taking pros of `tests/util.DBPath`
- Support `DBPath` option. They are mountpoints where user could put SST Files into

### CODE REVIEW

File: src/db_options.rs [hunk 1]
```diff
@@ -74,6 +74,94 @@ impl Drop for Cache {
     }
 }
 
+pub struct Env {
```
Comment (DarkEld3r) [inline src/db_options.rs:77]: Please add a doc-comment for this struct.

Comment (linxGnu) [reply src/db_options.rs:87]: I added description for `Env`. Thank you for quick reviewing :+1: @DarkEld3r

File: tests/test_db.rs [hunk 2]
```diff
@@ -79,16 +84,19 @@ fn writebatch_works() {
     {
         let db = DB::open_default(&path).unwrap();
         {
-            // test put
+            // test putx
```
[the remark this answers is not in the document]
Comment (linxGnu) [reply tests/test_db.rs:87]: ah just copy paste from old logic, let me remove this :+1:

[the remark this answers is not in the document]
Comment (linxGnu) [reply tests/test_db.rs]: Addressed your comment as well as make `cargo test` on document less error (conflicting data dir). PTAL, thank you :)

**Pull request merged**